src/Api/Request.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\commerce_canadapost\Api;
+
+use Drupal\commerce_canadapost\Plugin\Commerce\ShippingMethod\CanadaPost;
+use Drupal\commerce_store\Entity\StoreInterface;
+
+use Exception;
+
+/**
+ * CanadaPost API Service.
+ *
+ * @package Drupal\commerce_canadapost
+ */
+abstract class Request implements RequestInterface {
+
+  /**
+   * The logger channel factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Request class constructor.
+   */
+  public function __construct() {
+    $this->logger = \Drupal::service('logger.factory')->get(COMMERCE_CANADAPOST_LOGGER_CHANNEL);
```

Comment:
@thejacer87 @krystalcode Not sure why, but using the following:

```php
  /**
   * {@inheritdoc}
   */
  public static function create(ContainerInterface $container) {
    return new static(
      $container->get('logger.factory')
    );
  }
```
is giving me the following error and I'm not sure why: 

`ArgumentCountError: Too few arguments to function Drupal\commerce_canadapost\Api\Request::__construct(), 0 passed in /app/web/core/lib/Drupal/Component/DependencyInjection/Container.php on line 258 and exactly 1 expected in Drupal\commerce_canadapost\Api\Request->__construct() (line 41 of /app/web/modules/contrib/commerce_canadapost/src/Api/Request.php) #0 `

This is why I resorted to the `\Drupal::service` bit in the construct.

Reply:
I have tried that as well, and unfortunately, it doesn't work either.

Reply:
i can confirm the same issue. seems like create isn't being called. not sure why though

Reply:
@shabananavas You just need to add back the arguments to the services defined in `services.yml` that are removed as part of this PR. The service container passes the arguments to the constructor and that should be enough as long as we get the service from the container. It doesn't care if the constructor is inherited.

Give it another go and I'll have a look if you can't make it work.

Reply:
Okay, so, @krystalcode @thejacer87 this was a learning experience. So, when Dimitris was talking about adding the dependencies via the services.yml file, I actually did this:

```php
  commerce_canadapost.request_service:
    class: Drupal\commerce_canadapost\Api\RequestServiceBase
    arguments: ['@logger.factory', '@commerce_canadapost.utilities_service']
```

And that didn't work. Instead I guess we're supposed to inject the dependencies on the classes that extend the RequestServiceBase:

```php
  commerce_canadapost.rating_api:
    class: Drupal\commerce_canadapost\Api\RatingService
    arguments: ['@logger.factory', '@commerce_canadapost.utilities_service']

  commerce_canadapost.tracking_api:
    class: Drupal\commerce_canadapost\Api\TrackingService
    arguments: ['@logger.factory', '@commerce_canadapost.utilities_service']
```

So that worked. Again, this is my inexperience with PHP object oriented programming. Might be a good issue for a future training session.